intake_dal/tests/test_dal_online.py

```diff
@@ -27,33 +26,52 @@ def user_events_df():
     )
 
 
+@pytest.fixture
+def user_events_json():
+    return [{
+        "userid": 100,
+        "home_id": 3,
+        "action": "click",
+        "timestamp": {
+            "format": "DATETIME",
+            "time": "2012-05-01T00:00:00.000000000",
+        }
+    },
+        {
+            "userid": 101,
+            "home_id": 4,
+            "action": "click",
+            "timestamp": {
+                "format": "DATETIME",
+                "time": "2012-05-02T00:00:00.000000000",
+            }
+        }
+    ]
+
+
 @mock.patch("intake_dal.dal_online._http_put_avro_data_set")
 @mock.patch("intake_dal.dal_online._http_get_avro_data_set")
 def test_dal_online_write_read(
-    mock_get: MagicMock, mock_put: MagicMock, serving_cat: DalCatalog, user_events_df: pd.DataFrame
+        mock_get: MagicMock, mock_put: MagicMock, serving_cat: DalCatalog, user_events_df: pd.DataFrame,
+        user_events_json: List[Dict]
 ):
     canonical_name = "entity.user.user_events"
     avro_str = serialize_panda_df_to_str(
         user_events_df, schema=json.loads(serving_cat.metadata["data_schema"][canonical_name])
     )
-    mock_get.return_value = avro_str
+    mock_get.return_value = user_events_json
     mock_put.return_value = 200
 
     serving_cat.entity.user.user_events(storage_mode="serving").write(user_events_df)
     assert serving_cat.entity.user.user_events(key=100).read().iloc[0].userid == 100
     assert serving_cat.entity.user.user_events(key=100).read().iloc[0].home_id == 3
 
-    assert_frame_equal(user_events_df, serving_cat.entity.user.user_events(key=100).read(), check_dtype=False)
+    assert serving_cat.entity.user.user_events(key=100).read().iloc[1].userid == 101
+    assert serving_cat.entity.user.user_events(key=100).read().iloc[1].home_id == 4
 
     mock_get.assert_called()
     mock_put.assert_called()
 
-    assert_frame_equal(
-        user_events_df,
-        deserialize_avro_str_to_pandas(mock_put.call_args[0][1]["avro_rows"]),
-        check_dtype=False,
-    )
-
```

Comment:
The check on `mock_put.call_args[0][1]["avro_rows"]` ensures that intake-dal write is sending the correct avro to mock_put-> _http_put_avro_data_set.  Removing that line would remove that validation.  Ideally I should have made that a separate unit test.

Reply:
added check back in